app/src/main/java/woowacourse/movie/list/presenter/SettingPresenter.kt

```diff
@@ -0,0 +1,14 @@
+package woowacourse.movie.list.presenter
+
+import android.content.Context
+import woowacourse.movie.database.NotificationSharedPreference
+import woowacourse.movie.list.contract.SettingContract
+
+class SettingPresenter(
+    private val view: SettingContract.View
+) : SettingContract.Presenter {
+    override fun saveNotificationGranted(isGranted: Boolean) {
+        val sharedPreference = NotificationSharedPreference(view as Context)
+        sharedPreference.save(isGranted)
+    }
+}
```

Comment:
Presenter가 sharedPreference 객체를 주입받도록 하는건 어떨까요?
```kotlin
class SettingPresenter(
    private val view: SettingContract.View,
    private val sharedPreference: SharedPreference
) : SettingContract.Presenter {
```

또한, 이 주입받을 SharedPreference는 Application Context와 Activity Context 중 어느 Context에 의존해야 좋을지 고민해보시고 공유해주세요!

[참고해볼만한 글](https://stackoverflow.com/questions/11567134/sharedpreferences-application-context-vs-activity-context)

Reply:
링크 달아주신 글 잘 읽어보았습니다! 제이든의 이 코멘트를 처음 봤을 때에는 "아 Shared Preference는 모든 액티비티에서 공통으로 사용되니까 Application Context를 써야 하겠구나?" 라고 생각 했는데요, 그럼에도 불구하고 정상 동작했던 이유가 저 글에 들어있네요. 동일한 이름만 사용된다면 같은 인스턴스를 얻게 되는군요! 

하지만 그럼에도 불구하고 application context를 써야 하지 않을까 생각합니다. 앱 전체에서 공통으로 사용되니까요.

Reply:
shared preference를 프레젠터의 파라미터로 주입받도록 한다면 `(view as Fragment).requireContext().applicationContext`같이 길고 복잡하게 Applciation Context를 받아오지 않을 수 있을것 같군요! 하지만 문제가 존재했는데요.. 😭 fragment에서 생성자로 shared preference를 받을 시 부모 activity의 context가 필요한데, onAttach가 되지 않았기 때문에 부모의 context를 받아올 수가 없다고 합니다.. 이를 어떻게 해결하면 좋을까요..? 😥

Reply:
shared preference를 프레젠터의 파라미터로 주입받도록 한다면 (view as Fragment).requireContext().applicationContext같이 길고 복잡하게 Applciation Context를 받아오지 않을 수 있을것 같군요!

Reply:
맞습니다. Acitivty Context를 사용했지만 왜 정상적으로 동작했는지를 스스로 알아보기를 의도했는데 잘 수행해주셨네요 👍 

> So it actually does not matter which way you go, the important thing is to use the same name in order to get the same prefs from different parts of the application. However creating a single "access point" for the prefs could be a plus. So it could be a singleton wrapper over the prefs instantiated in Application.onCreate().

글 속의 이 내용을 공유드리고 싶었습니다. 내부구현으로 다행히 activity context으로도 잘 동작하지만, 거기서 끝나는게 아닌 왜 Application Context를 사용해야 하는지 알고 넘어가는게 중요하니까요.